pystac_client/client.py

```diff
@@ -94,6 +96,16 @@ def from_file(
 
         return cat
 
+    def _supports_collections(self) -> bool:
+        return self._conforms_to(ConformanceClasses.COLLECTIONS) or self._conforms_to(
+            ConformanceClasses.FEATURES
+        )
+
+    def _conforms_to(self, conformance_class: ConformanceClasses) -> bool:
+        if isinstance(self._stac_io, StacApiIO):
+            return self._stac_io.conforms_to(conformance_class)
+        return True
+
```

Comment:
Where did we land on "using `pystac-client` without a `StacApiIO`"? If we decided that it was the user's problem if they step outside that usage, then we should remove the `isinstance`.

Reply:
If someone did want to customize IO they should still inherit from StacApiIO, which would also return True for isinstance, so I think yes it's safe to remove the `isinstance` function. If we want to do any check it should be in the Client __init__ that asserts `self._stac_io` is an instance of StacApiIO.

Reply:
Makes me like @philvarner's `Protocol` suggestion to try to catch this sort of stuff in type checking?

Reply:
I just removed it and put in a TODO -- there's another branch from this branch with more changes that it will fit in better with. 

I think we landed on "use a StacApiIO instance"

Reply:
that's part of the types in a later commit